builder/src/main/java/dev/sashimono/builder/jar/JarTask.java

```diff
@@ -82,4 +92,18 @@ public void accept(Path file) {
         }
 
     }
+
+    private void writeManifestFile(final Path classesDir) throws IOException {
+        manifestEntries.put(BUILD_TOOL_JDK_SPEC, System.getProperty(JAVA_SPEC_VERSION));
+        final Path metaInfDir = classesDir.resolve("META-INF");
+        final Path manifestPath = metaInfDir.resolve(MANIFEST_MF);
+        Files.createDirectories(metaInfDir);
+        try (final BufferedWriter writer = Files.newBufferedWriter(manifestPath)) {
+            final String delimiter = ": ";
+            for (Map.Entry<String, String> entry : manifestEntries.entrySet()) {
+                writer.write(StringUtil.camelToCapitalisedKebabCase(entry.getKey()) + delimiter + entry.getValue()
```

Comment:
I think there are also some rules around max line length. Rather than writing it out manually JarOutputStream has a constructor that will take a `java.util.jar.Manifest` that will handle this formatting for you.

Reply:
https://docs.oracle.com/en/java/javase/22/docs/specs/jar/jar.html#notes-on-manifest-and-signature-files